### ISSUE TITLE
SC1134 ?!

### ISSUE THREAD

**Issue by milkpirate** [koalaman/shellcheck#2384]
#### For bugs
- Rule Id: SC1134
- My shellcheck version: 0.8.0 (snap)
- [x] The rule's wiki page does not already cover this: there is no wiki page for that rule ID.
![image](https://user-images.githubusercontent.com/9251021/143210959-5b5517d6-2248-4fbb-b3f3-05ce27377002.png)

- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
$ ls -la shellcheckrc 
664 -rw-rw-r-- 1 user user 1.3K Jan 27  2021 shellcheckrc
$ cat shellcheckrc 
# Look for 'source'd files relative to the checked script,
# and also look for absolute paths in /mnt/chroot
source-path=SCRIPTDIR
source-path=/

# Allow 'source' outside of FILES
external-sources=/opt/bin/text_lib.sh
external-sources=/opt/bin/shell_lib.sh

# Turn on warnings for unquoted variables with safe values
enable=quote-safe-variables

# Turn on warnings for unassigned uppercase variables
# enable=check-unassigned-uppercase

# Turn on warnings from sourced files
enable=check-sourced

# Minor settings
color=always

# Our busybox sh does have more functionality than the posix sh but is still
# not as mights as bash. So we check against posix sh and exclude warnings
# which are supported by the busybox sh.
shell=sh

# In POSIX sh, [[ ]] is undefined.
disable=SC3010

# In dash, [[ ]] is not supported.
disable=SC2169

# In POSIX sh, 'local' is undefined.
disable=SC2039

# In POSIX sh, string replacement is undefined.
disable=SC3060

# In POSIX sh, $'..' is undefined.
disable=SC3003

# As of 2019-09-08, BusyBox tr does not support character classes, so you
# would have to ignore this message (https://github.com/koalaman/shellcheck/wiki/SC2018#exceptions).
# Use '[:lower:]' to support accents and foreign alphabets.
disable=SC2018
# Use '[:upper:]' to support accents and foreign alphabets.
disable=SC2019
```

#### Here's what shellcheck currently says:

```sh
$ shellcheck /home/user/repos/repo/install.sh
In /home/user/repos/repo/install.sh line 3:
set -o xtrace
^-- SC1134 (error): Failed to process /home/user/repos/repo/shellcheckrc, line 7:  Fix any mentioned problems and try again.


In  /home/user/repos/repo/install.sh line 7:
[[ -d .odi ]] || cd "$(dirname "$0")/.."
^-----------^ SC3010 (warning): In POSIX sh, [[ ]] is undefined.
...
```

This is reported at any start of the any script. Always the first message.

#### Here's what I wanted or expected to see:

Not to see the message above an process the shellcheckrc (and output nothing since the file is clean with my rc).

```sh
$ shellcheck /home/user/repos/repo/install.sh
$
```

**Comment by milkpirate** [comment]
So?

**Comment by TinCanTech** [comment]
> So?

**So**, `shellcheck` did not configure itself to ignore 3010.

**Comment by milkpirate** [comment]
Well, but I did, via `schellcheckrc`:
```
# In POSIX sh, [[ ]] is undefined.
disable=SC3010
```
Sorry, but you have to be a little clearer and say a little more. I dont get the point you want to make.

**Comment by TinCanTech** [comment]
`shellchech` stopped processing `shellcheckrc` at line 7, so it did not read the rest of the file.

**Comment by milkpirate** [comment]
Ok, and why? The SC1134 is super unclear.

**Comment by TinCanTech** [comment]
Perhaps there is something about `/opt/bin/shell_lib.sh` .. ?

**Comment by milkpirate** [comment]
Ok, the wiki page you wrote wont help at all.

The files `/opt/bin/*_lib.sh` do not exit in this place because I am linting my scripts not on the target system where they run.

If I remove the line from `schellcheckrc` the linting explodes when sourcing. 
I dont want to ignore it on every occurence and I dont want to disable that check all together.

**Comment by milkpirate** [comment]
Ok thanks. Anyone else having helpful suggestions?

**Comment by Dmole** [comment]
@milkpirate  likely /opt/bin/shell_lib.sh is not a real file so comment it out of your shellcheckrc.

**Comment by koalaman** [comment]
This is indeed confusing. `external-sources` is a boolean value and only accepts `true`/`false` (there is currently no way to whitelist external files except by specifying them on the command line). The error message for this is missing, so you just get a reference to "mentioned problems" of which there are none. This definitely needs improving.

**Comment by milkpirate** [comment]
@Dmole yes, these files do not exit while running shellcheck. @koalaman In earlier versions (prior to 0.8) `external-sources` was was a witelisting for to-be-sourced scripts, wasnt it? Or did shellcheck in prior versions just treat the (non-empty) string as true?

Further, would it be possible to use `external-sources` as a white list (again)? Since I really dont want to allow to source everything and I dont want to explicitly allow it in every single occurrence of source and clutter my scripts.

**Comment by Mostafaviii** [comment]
> #### For bugs
> * Rule Id: SC1134
> * My shellcheck version: 0.8.0 (snap)
> * 0.7.2 in here the IDs (from 0.7.1) have changed. Its reading the shellcheckrc but it is then ignoreing the `disable=SC3043` in it.
> * 0.7.1 is working well
> * [x]  The rule's wiki page does not already cover this: there is no wiki page for that rule ID.
>   ![image](https://user-images.githubusercontent.com/9251021/143210959-5b5517d6-2248-4fbb-b3f3-05ce27377002.png)
> * [x]  ~I tried on https://www.shellcheck.net/ and~ verified that this is still a problem on the latest commit
> 
> #### Here's a snippet or screenshot that shows the problem:
> ```shell
> $ ls -la shellcheckrc 
> 664 -rw-rw-r-- 1 user user 1.3K Jan 27  2021 shellcheckrc
> $ cat shellcheckrc 
> # Look for 'source'd files relative to the checked script,
> # and also look for absolute paths in /mnt/chroot
> source-path=SCRIPTDIR
> source-path=/
> 
> # Allow 'source' outside of FILES
> external-sources=/opt/bin/text_lib.sh
> external-sources=/opt/bin/shell_lib.sh
> 
> # Turn on warnings for unquoted variables with safe values
> enable=quote-safe-variables
> 
> # Turn on warnings for unassigned uppercase variables
> # enable=check-unassigned-uppercase
> 
> # Turn on warnings from sourced files
> enable=check-sourced
> 
> # Minor settings
> color=always
> 
> # Our busybox sh does have more functionality than the posix sh but is still
> # not as mights as bash. So we check against posix sh and exclude warnings
> # which are supported by the busybox sh.
> shell=sh
> 
> # In POSIX sh, [[ ]] is undefined.
> disable=SC3010
> 
> # In dash, [[ ]] is not supported.
> disable=SC2169
> 
> # In POSIX sh, 'local' is undefined.
> disable=SC2039
> 
> # In POSIX sh, string replacement is undefined.
> disable=SC3060
> 
> # In POSIX sh, $'..' is undefined.
> disable=SC3003
> 
> # As of 2019-09-08, BusyBox tr does not support character classes, so you
> # would have to ignore this message (https://github.com/koalaman/shellcheck/wiki/SC2018#exceptions).
> # Use '[:lower:]' to support accents and foreign alphabets.
> disable=SC2018
> # Use '[:upper:]' to support accents and foreign alphabets.
> disable=SC2019
> ```
> 
> #### Here's what shellcheck currently says:
> ```shell
> $ shellcheck /home/user/repos/repo/install.sh
> In /home/user/repos/repo/install.sh line 3:
> set -o xtrace
> ^-- SC1134 (error): Failed to process /home/user/repos/repo/shellcheckrc, line 7:  Fix any mentioned problems and try again.
> 
> 
> In  /home/user/repos/repo/install.sh line 7:
> [[ -d .odi ]] || cd "$(dirname "$0")/.."
> ^-----------^ SC3010 (warning): In POSIX sh, [[ ]] is undefined.
> ...
> ```
> 
> This is reported at any start of the any script. Always the first message.
> 
> #### Here's what I wanted or expected to see:
> Not to see the message above an process the shellcheckrc (and output nothing since the file is clean with my rc).
> 
> ```shell
> $ shellcheck /home/user/repos/repo/install.sh
> $
> ```
